source/particle/particle.cc

```diff
@@ -76,11 +76,47 @@ namespace aspect
       data = static_cast<const void *> (pdata);
     }
 
+#ifdef DEAL_II_WITH_CXX11
+    template <int dim>
+    Particle<dim>::Particle (const Particle<dim> &particle)
+      :
+      location (particle.location),
+      reference_location(particle.reference_location),
+      id (particle.id),
+      properties(particle.properties)
+    {}
+
+    template <int dim>
+    Particle<dim>::Particle (Particle<dim> &&particle)
+      :
+      location (particle.location),
+      reference_location(particle.reference_location),
+      id (particle.id),
```

Comment:
If this is a non-trivial data object, you probably want to also use `std::move(particle.id)` here. This may require the addition of a move constructor to `CellId`.